lib/puppet/node.rb

```diff
@@ -23,8 +23,8 @@ class Puppet::Node
 
   def initialize_from_hash(data)
     @name = data['name']
-    @classes = data['classes']
-    @parameters = data['parameters']
+    @classes = data['classes'] || []
+    @parameters = data['parameters'] || {}
     @environment_name = data['environment']
```

Comment:
Hm, actually, what I actually will do is to move the check, since initialize_from_hash is the rock bottom, not self_from_data_hash.